### PR TITLE
KZN-2614-convert-remaining-jest-tests

### PR DESCRIPTION
**Pull request by gyfchong**
Converted the remaining jest reliant tests to either vitest or storybook where relevant

### CODE REVIEW

File: packages/components/src/Illustration/subcomponents/VideoPlayer/VideoPlayer.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    Object.defineProperty(HTMLMediaElement.prototype, "muted", {
+      set: vi.fn(),
+    })
+  })
+
+  afterEach(() => {
+    vi.clearAllMocks()
+  })
+
+  it("renders a video player in the document and autoplay", async () => {
```
Comment (HeartSquared) [inline packages/components/src/Illustration/subcomponents/VideoPlayer/VideoPlayer.spec.tsx:45]:
```suggestion
  it("renders a video player in the document and autoplay", () => {
```

File: packages/components/src/Illustration/subcomponents/VideoPlayer/VideoPlayer.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  title: "Components/Illustrations/Tests",
+  parameters: {
+    chromatic: { disable: false },
+    controls: { disable: true },
+  },
+} satisfies Meta
+
+const StickerSheetTemplate: StickerSheetStory = {
+  render: ({ isReversed }) => (
+    <StickerSheet isReversed={isReversed} title="Aspect Ratio">
```
Comment (gyfchong) [inline packages/components/src/Illustration/subcomponents/VideoPlayer/VideoPlayer.stories.tsx:20]:
```suggestion
  render: () => (
    <StickerSheet title="Aspect Ratio">
```

File: packages/components/src/Illustration/subcomponents/VideoPlayer/VideoPlayer.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import { Meta } from "@storybook/react"
+import { userEvent, within, expect, waitFor } from "@storybook/test"
+import {
+  StickerSheet,
+  StickerSheetStory,
+} from "~storybook/components/StickerSheet"
+import { VideoPlayer } from "./index"
+
+export default {
+  title: "Components/Illustrations/Tests",
```
Comment (HeartSquared) [inline packages/components/src/Illustration/subcomponents/VideoPlayer/VideoPlayer.stories.tsx:11]: Given this isn't actually a `stickersheet.stories.tsx` file, the meta should follow the normal stories format.

```suggestion
  title: "Components/Illustrations/VideoPlayer",
```

File: packages/components/src/Notification/subcomponents/GenericNotification/GenericNotification.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import React, { useState } from "react"
+import { Meta } from "@storybook/react"
+import { userEvent, within, expect, waitFor } from "@storybook/test"
+import { StickerSheetStory } from "~storybook/components/StickerSheet"
+import { GenericNotification } from "./index"
+
+export default {
+  title: "Components/Notifications/Tests",
```
Comment (HeartSquared) [inline packages/components/src/Notification/subcomponents/GenericNotification/GenericNotification.stories.tsx:8]: Something like this
```suggestion
  title: "Components/Notifications/Generic Notification",
```

File: packages/components/src/Notification/subcomponents/GenericNotification/GenericNotification.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  parameters: {
+    chromatic: { disable: false },
+    controls: { disable: true },
+  },
+} satisfies Meta
+
+const StickerSheetTemplate: StickerSheetStory = {
```
Comment (HeartSquared) [inline packages/components/src/Notification/subcomponents/GenericNotification/GenericNotification.stories.tsx:15]: Not really a stickersheet, so maybe just move the render into the test story

File: packages/components/src/Text/Text.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  describe("changes rendered HTML element when passed tag", () => {
+    const testCases: TextVariants[] = [
+      "intro-lede",
+      "body",
+      "small",
+      "extra-small",
+    ]
+
+    testCases.forEach(variant => {
+      it(`renders the correct element for <Text variant={${variant}} />`, () => {
```
Comment (HeartSquared) [inline packages/components/src/Text/Text.spec.tsx:15]: This test is for the `tag` prop, the `variant` makes no difference.

File: packages/components/src/Text/Text.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+          <Text variant={variant} tag="div">
+            Example
+          </Text>
+        )
+        expect(textMock.getByText("Example").tagName).toBe("DIV")
+      })
+    })
+  })
+
+  it("passes through data attributes", () => {
```
Comment (HeartSquared) [inline packages/components/src/Text/Text.spec.tsx:26]: Maybe we should just delete this test 😬

File: packages/components/src/Text/Text.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+      <Text variant="intro-lede" data-testid="test-id">
+        Example
+      </Text>
+    )
+    expect(getByTestId("test-id")).toBeInTheDocument()
+  })
+
+  describe("defaults to the correct HTML element", () => {
+    type TestObject = { variant: TextVariants; el: AllowedTextTags }
+    const testCases: TestObject[] = [
```
Comment (HeartSquared) [inline packages/components/src/Text/Text.spec.tsx:37]: Everything is just a p tag? 😅 One test is enough

**Pull request merged**